examples/deep_clone_schema.py

```diff
@@ -0,0 +1,124 @@
+# MAGIC
+# MAGIC This notebook serves as an example of utilizing DiscoverX to automate the replication of a schema using Delta Deep Clone.
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## [Optional] Create source and destination for the example
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC create catalog if not exists _discoverx_deep_clone;
+# MAGIC create schema if not exists _discoverx_deep_clone.samples;
+# MAGIC use catalog _discoverx_deep_clone;
+# MAGIC use schema samples;
+# MAGIC
+# MAGIC CREATE OR REPLACE TABLE num_table
+# MAGIC   (id BIGINT GENERATED BY DEFAULT AS IDENTITY, num INT);
+# MAGIC
+# MAGIC INSERT INTO num_table(num)
+# MAGIC SELECT explode(sequence(1, 200, 2));
+# MAGIC
+# MAGIC CREATE OR REPLACE TABLE date_table (
+# MAGIC   id BIGINT GENERATED BY DEFAULT AS IDENTITY,
+# MAGIC   date_col DATE
+# MAGIC );
+# MAGIC
+# MAGIC INSERT INTO date_table(date_col)
+# MAGIC SELECT explode(sequence(CAST(DATE'2022-01-01' AS TIMESTAMP), CAST(DATE'2023-12-01' AS TIMESTAMP), INTERVAL 1 MONTH));
+# MAGIC
+# MAGIC CREATE OR REPLACE VIEW num_view AS SELECT * FROM num_table;
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Specify a source and a destination catalog
+
+# COMMAND ----------
+
+dbutils.widgets.text("1.source_catalog", "_discoverx_deep_clone") 
+dbutils.widgets.text("2.destination_catalog", "_discoverx_deep_clone_replica")
+
+source_catalog = dbutils.widgets.get("1.source_catalog")
+destination_catalog = dbutils.widgets.get("2.destination_catalog")
+
+# COMMAND ----------
+
+print(f"Replicate data from {source_catalog} to {destination_catalog} ")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Use DiscoverX for cloning
+
+# COMMAND ----------
+
+# %pip install dbl-discoverx==0.0.7
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+
+from discoverx import DX
+
+dx = DX()
+
+# COMMAND ----------
+
+spark.sql(f"CREATE CATALOG IF NOT EXISTS {destination_catalog}")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Define a function for cloning
+
+# COMMAND ----------
+
+def clone_tables(table_info):
+  
+  spark.sql(f"CREATE SCHEMA IF NOT EXISTS {destination_catalog}.{table_info.schema}")
+  
+  try:
+    spark.sql(
+    f"""CREATE OR REPLACE TABLE 
+    {destination_catalog}.{table_info.schema}.{table_info.table} 
+    CLONE {table_info.catalog}.{table_info.schema}.{table_info.table}
+    """
+  )
+    info = f"Cloned {table_info.catalog}.{table_info.schema}.{table_info.table} to {destination_catalog}.{table_info.schema}.{table_info.table}" 
+  # Cloning Views is not supported
+  except Exception as error: 
+    info = f"{error}  Cloning Skipped "
```

Comment:
This is an iteresting information, it would be great to structure the resoponse instead of a string to have the structured information (succes (true/false), catalog name, etc.).
This way if you have more than one error you should probably throw an exception at the end of the notebook.

Reply:
reformatted the output as a dict that